x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/DocumentPermissions.java

```diff
@@ -240,49 +238,44 @@ public static DocumentPermissions allowAll() {
      * @return instance of {@link DocumentPermissions}
      */
     public DocumentPermissions limitDocumentPermissions(DocumentPermissions limitedByDocumentPermissions) {
-        assert limitedByQueries == null && limitedByDocumentPermissions.limitedByQueries == null
-            : "nested scoping for document permissions is not permitted";
-        if (queries == null && limitedByDocumentPermissions.queries == null) {
+        if (hasDocumentLevelPermissions() && limitedByDocumentPermissions.hasDocumentLevelPermissions()) {
+            return new DocumentPermissions(
+                Stream.concat(getListOfQueries().stream(), limitedByDocumentPermissions.getListOfQueries().stream()).toList()
```

Comment:
Is this logically the same to smash to these two together ? e.g. is there a scenario where treating them as separate lists would result in different behavior ?

Reply:
We sync'd offline. The operation here concatenate two lists of sets (List<Set<>>). It does not change anything within the Set.

Reply:
:+1: